test/e2e/aks.go

```diff
@@ -355,6 +355,68 @@ func GetLatestStableAKSKubernetesVersion(ctx context.Context, subscriptionID, lo
 	return maxVersion, nil
 }
 
+type AKSMachinePoolSpecInput struct {
+	Cluster         *clusterv1.Cluster
+	MachinePools    []*expv1.MachinePool
+	InitialReplicas int32
+	WaitIntervals   []interface{}
+}
+
+func AKSMachinePoolSpec(ctx context.Context, inputGetter func() AKSMachinePoolSpecInput) {
+	input := inputGetter()
+
+	By("Scaling the machine pools out")
+	framework.ScaleMachinePoolAndWait(ctx, framework.ScaleMachinePoolAndWaitInput{
+		ClusterProxy:              bootstrapClusterProxy,
+		Cluster:                   input.Cluster,
+		Replicas:                  input.InitialReplicas + 1,
```

Comment:
Seems like the scaling test failed which is interesting of itself, but I would echo my comments on the prior E2E PR that rather than take an `InitialReplicas` as input let's just get the input replicas count in real-time by getting the machinepool resource and then adding 1 to that for the scale out value.

Reply:
Yeah that makes sense. And since this test scales all of the machine pools, do you think it's also worth removing the assumption that all machine pools have the same number of replicas? Or should we only scale one machine pool at a time?

Reply:
I think AKS only allows one node pool scaling event at a time (that assumption should be validated), in which case we'll have to serialize.

Yeah, I think we just enumerate through all pools in the cluster, scale out by <whatever>, validate, then scale back in, validate. Ideally we can do those all concurrently in discrete goroutines to economize for E2E duration, but as I suggested above that might not be possible.

Reply:
When I was first testing this, scaling all the machine pools at once seemed to work. I can see where iterating over each one would make it easier to more granularly tweak replica counts, and that would probably be simple enough to wrap in a loop and do in parallel.